### ISSUE TITLE
Using Transifex for localization of qtranslate-x

### ISSUE THREAD

**Issue by bekatoros** [qTranslate-Team/qtranslate-x#72]
Hi, I want to propose the use of https://www.transifex.com for the localization of the plugin. It is free for open source projects.

**Comment by johnclause** [comment]
Do you mean to translate -X content itself, or to translate client content? I am open to both, I just do not have time to deal with it right now. Do you wish to take the lead and to make it work?

**Comment by bekatoros** [comment]
I can make a try.

**Comment by johnclause** [comment]
It would be great. Which one:
>Do you mean to translate -X content itself, or to translate client content?

**Comment by pedro-mendonca** [comment]
I use Transifex for some wordpress projects, it have a good UI and some usefull tools. At the moment it's more complete than glotpress.

**Comment by bekatoros** [comment]
I have set up the project using the POT and PO files https://www.transifex.com/projects/p/qtranslate-x/ 
you can make a request to join and i will accept as soon as possible.

**Comment by pedro-mendonca** [comment]
There is a comunity of WP-Translations, a lot of projects are there:
https://www.transifex.com/organization/wp-translations/

**Comment by pedro-mendonca** [comment]
More on that here:
http://wp-translations.org/

**Comment by pedro-mendonca** [comment]
I relly a lot on poedit to translate and debug some string problems, but from the author point of view, putting the plugin in the spotlight at a community like wp-translations will surely bring a lot more people to complete language translations, probably way more than having qtranslate itself as an isolated project on transifex.
@johnclause you should consider managing a close dedicated team for qtranslate plugin and other related plugins, or register qtranslate-x here: http://wp-translations.org/developers-wp-translations/ for a more wide community of translators.

**Comment by bekatoros** [comment]
@johnclause when you make a request to join I will add you as an administrator, just sent me the username if it is different .

**Comment by qtranslateteam** [comment]
I cannot find a button "join" so far. username: 'qtranslateteam'

**Comment by qtranslateteam** [comment]
Are they going to translate for free? I wonder how that can be possible without seeing a context?

**Comment by bekatoros** [comment]
Of course, you can add it as a link at the plugin readme, I have added you as a administrator.

**Comment by qtranslateteam** [comment]
Is there a way to add qtranslateteam to Maintainers on page https://www.transifex.com/projects/p/qtranslate-x/?  Are you, @bekatoros, going to translate some?

**Comment by johnclause** [comment]
Pedro (@pedro-mendonca), would you prefer to use it over poedit?

Marios, is there a way to upload new .pot file from a linux script? And then from a script, also check what new translations are available, and download them?

**Comment by bekatoros** [comment]
Yes i started the greek localization. They have an API but I dont know if some has created a script. http://docs.transifex.com/developer/api/resources

**Comment by johnclause** [comment]
The "Details" section shows source line, which I was missing at first place. It seems reasonable to me, but it is up to translators to say if they want to use it. I guess, we already have a vote from Marios (@bekatoros)? I think I will figure out a way to script it, if translators say they want to use it.

How sure are you that it will stay free after 30 days: https://www.transifex.com/pricing/ ?

**Comment by pedro-mendonca** [comment]
I'm pretty ok with Poedit, it works like a charm.
At the speed that this qTranslate-X is being developed I really like more the Poedit system because I can sweep directly the code and check context of strings. This is much of a help in case of debugging strings and so forth. About the price of Transifex, don't know anything about it, I use it mostly for translating other users stuff, the price is on their side. Still, don't know if it would cost anything if registred within wp-translations-team, as they are the owners of the account, and rely on volunteer work to translate wordpress registered themes and plugins projects.
Even if qtranslate is going to Transifex or self hosted Glotpress, for now, as I've explained above, I'll keep for my personal use the Poedit method, at least while I'll keep finding bugs and suggesting features, as I like to keep an eye on the code. In the future, when qtranslate-x enters cruise speed and slows a bit the development, I might just translate whatever new strings keep appearing, without worries and with no need to step in the strings syntax etc.
One more point, I think Transifex looks great from a translators point of view, but have no idea how it's kept synced with the source code, and in the last weeks a lot keeps changing every day, I like watching the code in github and update language files directly on it. Still, this is an issue that over the time will stabilize, as qtranslate gets more and more mature.
One experience I have with Glotpress self hosted projects for translation, the admin sends some emails to alert translators that there are new strings to translate. Don't know how that works within Transifex admin dashboard. In GitHub we just keep receiving update notifications directly, and just have to check in Poedit if there are any new untranslated strings waiting.
In resume, I'm ok with how it is now, I will use other system if you'll implement it, but I'm keeping Poedit for a while. Still, it's a good idea to check these other possibilities.

**Comment by Grafcom** [comment]
There's something I noticed in Transifex. If anything has changed in the (pre-existing) original text, then the box to translate is empty. Some long texts must then be completely retranslated. In PoEdit is the previous translation still visible (fuzzy) and easy to adjust again. But maybe I missed something.

**Comment by pedro-mendonca** [comment]
That's right, Poedit shows a close old string with #fuzzy. Sometimes is too different and I still pick it up opening .po in notepad.

**Comment by johnclause** [comment]
This seems to be a big one ... I was also looking for 'fuzzy' and did not see it in transifex, but I did not realized that it is even worse. They completely remove all commented translations.

Well, it does not hurt to have both ways for now and let people to decide individually. Marios, do you have experience with poedit to be able to compare?

**Comment by pedro-mendonca** [comment]
Same problem with Glotpress, both platforms give up the control you have in local files via Poedit.

**Comment by Grafcom** [comment]
A "simple" instruction for using Poedit in combination with GitHub would be very useful for many translators :-)

**Comment by pedro-mendonca** [comment]
I've discussed that in a sort of looong issue with @johnclause, maybe it's time to make a wiki page on translation. :)

**Comment by Grafcom** [comment]
@pedro-mendonca,
That would be totally awesome :-)  :-)  :-)

**Comment by bekatoros** [comment]
I personally have also used the poedit but I find it more difficult for translating.

**Comment by Grafcom** [comment]
@bekatoros,

Can you explain to me why this is more difficult, so I understand what you mean?

**Comment by johnclause** [comment]
Could someone take initiative and word such a document out? Also please suggest where it would be better to publish it, https://qtranslatexteam.wordpress.com/about/, or GitHub wiki, or anywhere else?

**Comment by pedro-mendonca** [comment]
I can compile everything we dissussed before and write it to a wiki for start.
As we are using GitHub to pull language files, I think this might be the right place to put this information.
After it's done and reviewed by you, maybe you can include some link to it in the blog.
Another suggestion I have is divide all the info on readme to some more pages, as it is getting bigger every day.
In a few minutes I'll send you my first step suggestion.

**Comment by pedro-mendonca** [comment]
This is my start suggestion, it's basically the same readme but with some markdown according GitHub rules to improve the rendering of readme in the fron page.
https://github.com/qTranslate-Team/qtranslate-x/pull/73

Keeping both .txt and .md is redundant. Maybe keeping both is helpfull for different contexts as .md is very helpfull for GitHub render readme a lot better, .txt is the probably the best for one who reads it locally after plugin download.
To keep it simple, might be helpfull to just make them both with same markdown.

**Comment by pedro-mendonca** [comment]
As can be read here:
https://help.github.com/articles/about-github-wikis/
"GitHub Wikis are a place in your repository where you can share long-form content about your project, such as how to use it, how it's been designed, manifestos on its core principles, and so on. Whereas a README is intended to quickly orient readers as to what your project can do, wikis can be used to provide additional documentation."

@johnclause what do you think of migrating some of the actual too long readme to some organized pages?
If so, which ones would you suggest? i would go for FAQ, Features, Installation and the Translation How To that we've been discussing.

**Comment by pedro-mendonca** [comment]
I've started the paste here a lot of stuff from readme.
https://github.com/qTranslate-Team/qtranslate-x/wiki

The translation procedures will come here.
https://github.com/qTranslate-Team/qtranslate-x/wiki/Translation
@johnclause what do you think of squeezing the necessary info from herehttps://github.com/qTranslate-Team/qtranslate-x/issues/3?

**Comment by johnclause** [comment]
It seems to me that from simple user point of view this site https://qtranslatexteam.wordpress.com looks much better and it also related to wordpress, which make people feel comfortable. GitHub is better for techy people and for developers.

I would keep tech instructions and documentation for translators at GitHub, since as you correctly noted, this is where translators will have to come anyway, while non-tech stuff probably better to live at https://qtranslatexteam.wordpress.com, and we will crosslink both places. What do you think?

Let us populate https://github.com/qTranslate-Team/qtranslate-x/wiki/Translation page as the next priority, and once it is ready I will link it from everywhere where it is possible. Is that a good plan?

**Comment by Grafcom** [comment]
I completely agree with John and @pedro-mendonca  nice work so far!

**Comment by pedro-mendonca** [comment]
I agree.
I just created this pages to introduce some text before the Translation one, and thought that copying from readme would make sense so that the text on that process is not isolated on the wiki.
I used this also as a test to understand how to create wikis and sidebar.
Of course you should centralize everything, and the Website is a good option.

I will take translation stuff and start the wiki with the conclusions we came on the early issue.

**Comment by johnclause** [comment]
Marios (@bekatoros), when you are done with greek translation, please, let me know, since I am not sure if I get an automated message from transifex for now. I hope they will not lock your work after 30 days asking for payment ... Although, it would be even better, if you submit pull request with your updated .po file on github, but any way is good - I most appreciate that you are doing the hardest part - translation itself.

**Comment by johnclause** [comment]
Pedro, I saw your pages, looks great. I edited a little bit other pages, which you created as examples, we need to make sure that there is no duplicated content. I will tidy it up later.

I linked your "Guide to Translation" from https://qtranslatexteam.wordpress.com/translators/ page.

What would be also good is to have for new translators step-by-step specific instructions on how to fork, clone, edit, update and submit pull request. A few people asked me for this information, and it would be great, if we could simply advise to follow some brainless steps, so that people would not have to search web on how to do it. When it is done at least once, then it is easy, but initial learning is a trouble for most of the people, even if they sincerely want to learn.

BTW, Pedro (@pedro-mendonca) and Marios (@bekatoros), if you have some site to advertise, I can put it on sponsor page https://qtranslatexteam.wordpress.com/credentials/, if it would have any value to you? Write to me through https://qtranslatexteam.wordpress.com/contact-us/. Gunu (@Grafcom) is already there ;)

Thanks!

**Comment by pedro-mendonca** [comment]
Hi

I've started the guide but not finished yet.
The only page that is more or less complete is the one with the best practices to translation. The main page have some headings just to prepare the sketch, but still have no content. I was thinking in two methods, one for tech users who can use GitHub (we can make a simple guide on how to), another option would be by email because I think there are users that can make a great translation work without having the techy barrier of github. In this case, they can simple send you the .po and you would take care of the merge itself. What do you think!
Next, I would separate guides to kick start a language translation and to update an existing one.
About a link to a website, let me think about it :)
Thanks

**Comment by johnclause** [comment]
>can simple send you the .po

Yes, e-mail is fine, but even they need step instructions how to get __the latest__ file from GitHub to translate at first place.

Problem with e-mail is when two people do updates at the same time, which already actually happened with de_DE. Merging is not easy, unless you use GitHub client. I would emphasise that GitHub is preferable option in long run, while e-mail is good for the start. Does this sound reasonable?

Thanks for everything!

**Comment by pedro-mendonca** [comment]
Yes, I agree.
I'll continue this as soon as I can.

**Comment by johnclause** [comment]
Marios (@bekatoros), can you submit, whatever you already got for translation, since we are about to release a new version of q-X? You do not have to finish it all, something is better than nothing ;) Please, also do not forget to merge your po with the latest from github.

**Comment by johnclause** [comment]
Marios, @bekatoros, we also do not have 'el' in the list of predefined configurations. Could you give me the values of all the fields, as you enter them on page '/wp-admin/options-general.php?page=qtranslate-x&edit=el'?

**Comment by Grafcom** [comment]
@pedro-mendonca ,
it is possible to explain here  - https://github.com/qTranslate-Team/qtranslate-x/wiki/Guide-to-Translation - how translators can do this via github?

**Comment by bekatoros** [comment]
Language Code : el
Name : Greek
Locale : el_GR
Not Available Message : Δεν υπάρχει μετάφραση

**Comment by johnclause** [comment]
Thanks, Marios, @bekatoros! I also need Date Format and Time Format? Please, figure out not empty ones, I cannot have blank in predefined language. You may look through examples of other languages in function `qtranxf_default_date_format`: https://github.com/qTranslate-Team/qtranslate-x/blob/master/qtranslate_options.php#L250 and function `qtranxf_default_time_format`: https://github.com/qTranslate-Team/qtranslate-x/blob/master/qtranslate_options.php#L284

"Name" must be spelled in native language, I do no believe it is "Greek" ;)

"Not Available Message" must be of the form:
>Sorry, this entry is only available in %LANG:, : and %.

It must contain %LANG, etc., please, read help text and see the examples of other not available messages in function `qtranxf_default_not_available` in file qtranslate_options.php: https://github.com/qTranslate-Team/qtranslate-x/blob/master/qtranslate_options.php#L213. Apparently you never saw how this message actually works.

**Comment by bekatoros** [comment]
I edited my comment and edited the configuration file

**Comment by johnclause** [comment]
Thanks, Marios!

**Comment by pedro-mendonca** [comment]
@Grafcom 
Yes, I'll do it in the next few days.

**Comment by pedro-mendonca** [comment]
@johnclause I've renamed the *Guide to Translation* to *[Guide to Plugin Translation](https://github.com/qTranslate-Team/qtranslate-x/wiki/Guide-to-Plugin-Translation)* to avoid the confusion between Translation of content vs plugin itself, as this is a translation plugin.
If you agree, please don't forget to update it's name and link in the page [Translators](https://qtranslatexteam.wordpress.com/translators/) page in the main qTranslate Team site.

Next, I will do the rest of the missing `IN PROGRESS` contents, It might be usefull to open a new issue apart from this one to discuss how things can be done ore improved in the wiki.

**Comment by johnclause** [comment]
I updated the link  - thank you, Pedro, for all your help :+1:

**Comment by johnclause** [comment]
@bekatoros: can you still use transifex? I tried to go there, it looks ok, but when I tried to pretend doing actual translation, it does not work. Is it because we have to pay?

**Comment by johnclause** [comment]
Is anyone still interested in transifex? I have hard time to figure out how to upload/download programmatically. Their documentation is not user friendly, at least not to me ;)

**Comment by johnclause** [comment]
I've figured out how to script it, but nobody seems to use it anyway, including @bekatoros, who started this thing. I am closing this issue for now, we can still write into a closed issue, or we can re-open it, if needed.

**Comment by bekatoros** [comment]
Yes it is a while since I translated anything! Been quite busy.

**Comment by pedro-mendonca** [comment]
@johnclause there is new situation that should now be considered, the WordPress translate site.
https://translate.wordpress.org/locale/pt/default/wp-plugins/qtranslate-x

Translating teams can now translate this plugin on WordPress portal, if people keep translating both on github, transifex and now in this official WP translation site, wich ones will be chosen?

I think that some developers opt to not put tha plugin to translate in this new site, and keep their translating teams that are already close to them. Some of them use a personal GlotPress install, also a good situation, maybe better than pure github that need a bit of dev skils.

Relying on this new WP site makes it dependent of current translating teams for each language, and some cases are completely different people and workflow from the current qtranslate-x translators. In my case it's the same, I can translate in both situations, but I believe this is not the most common case.

This is a matter that should be considered now.

**Comment by johnclause** [comment]
@bekatoros: I am glad you are alive :smile: Are you still using qtx on any of the sites?

Yes, there are now four ways (including e-mail). So far it that has not been a problem each translator does it one way and normally there is no conflict, but it may become a problem at some point.

I would think that we need to take advantage of WP way, as the most native way now. I definitely do not want to do GlotPress install, as I am trying to minimize maintenance as much as possible, and this would be a huge new piece to take care of, which would duplicate WP, transifex, etc. I would think that it is an old way people did when other tools were not available.

I assume WP automates translation updates on plugin too. I have not learned yet how they work. Do they do source parsing on their own? Do they allow to specify keywords, for example? Where do they take po files from? I currently removed po files from svn, as I thought we only need them on GitHub. Now it may change.

**Comment by bekatoros** [comment]
Yes in 3 now I think.

**Comment by Grafcom** [comment]
@pedro-mendonca and @johnclause 
I also wanted to look at the WordPress method. My question is especially who manages the translators / translations?
User Roles and Permissions - Guest - Contributor - Validator...
Who is who here? Can you set up a validator for each language? etc. etc.

**Comment by pedro-mendonca** [comment]
Exactly, that is the main problem I've mentioned above, the workflow and the teams aren't necessarily the same. I've seen in other plugins some developers asking translation teams to include their regular translators in the their translation teams. This would depend a lot on the teams. For pt_PT that is not a point, as I'm a validator, I can both do it on GitHub and WP.
On WP there is another issue, I believe there is no mechanism to notify translators of new strings to be translated. But as John mentioned, I think that the language files are somehow already being updated apart from the regular release cycle of the plugins.

**Comment by Grafcom** [comment]
@pedro-mendonca 
for my clarity, as co author of the plugin where can I find the settings in WP for the translations of qTranslate X ? I have not found it yet....

**Comment by pedro-mendonca** [comment]
I don't know, I only use the translation WordPress site as a translator/validator.
https://translate.wordpress.org/locale/pt/default/wp-plugins/qtranslate-x

**Comment by Grafcom** [comment]
@pedro-mendonca 
Do I understand you correctly? This is the team for Português / Portuguese (Portugal) 
https://make.wordpress.org/polyglots/teams/?locale=pt_PT
You all can validate the translation for qTranslate x (and other plugins) in Portuguese ?
The author of the plugin is not notified of changes?

**Comment by pedro-mendonca** [comment]
Yes, this is the pt_PT Team.
The validator permission is per team, the permission allows to validate translations of core, themes, plugins, etc.
I don't know the workflow after validation, don't even know if the auhor is notified. I think that in a near future (if not now already) the WordPress will update the .mo files for themes and plugins as it do for the core. Making the translations independente from the projects release cycle.

**Comment by johnclause** [comment]
>Making the translations independent from the projects release cycle.

I like that. Still have to find time to learn how they did it ...

**Comment by pedro-mendonca** [comment]
There is already translations updates on the update core panel, I've seen it for the core themes, twelve, thirteen, etc. I think it will be the same for themes and plugins. Probably it will be obtained from the translation repository itself.

**Comment by Grafcom** [comment]
Did a little test - Date added: 2015-10-11 16:41:54 GMT
> There is still: Status: waiting 

Waiting for whom?

**Comment by pedro-mendonca** [comment]
You did this "Beoordeel Gunu opie %s"

It can only be validated by users from the translation team of the correspondent language.

If you don't have validator permissions you should ask your language team either to validate or if you're used to translate maybe they can include you.

**Comment by pedro-mendonca** [comment]
A nicer link to check out qtranslate-x in your language:
https://translate.wordpress.org/locale/nl/default/wp-plugins/qtranslate-x

**Comment by Grafcom** [comment]
so I must become a member of the Dutch translation team?

**Comment by johnclause** [comment]
I would try reading WP [Translator Handbook](https://make.wordpress.org/polyglots/handbook/tools/glotpress-translate-wordpress-org/) all the way through, as it is advised on the entry page https://translate.wordpress.org/projects/wp-plugins/qtranslate-x, but I do not have time for that right now.

**Comment by Grafcom** [comment]
But if I have to be a member of the WP translation team then I do not need to read it all... :-)
Fortunately there are already many members. Translation Editors and Translation Contributors.
If this is the only way in the future, then this plugin is going to lose a translator... Sorry john.

**Comment by johnclause** [comment]
Well, I saw the current task now as to understand how it works before making any decisions. I was hoping that someone may go through all of this and explain us in two words what needs to be done, if anything. I am not quite sure how WP did it. There is a team of translators, how would they know that qtx needs translation? There are thousands of plugins, do they expect one team to translate them all? How many members in that team? How many validators? I would then think it should be thousands of them? Is there a grouping by plugins? Can you become a validator and then choose to work on qtx only?

**Comment by Grafcom** [comment]
> There is a team of translators, how would they know that qtx needs translation? There are thousands of plugins, do they expect one team to translate them all? How many members in that team? How many validators? I would then think it should be thousands of them? Is there a grouping by plugins? Can you become a validator and then choose to work on qtx only?

That is also the point.
Questions I have, I do not find answers in this handbook it just raises more questions...
To translate (core) WordPress in your language it mainly concerns.
Yes, I have not read it all but I scan the chapters...

https://make.wordpress.org/polyglots/handbook/about/teams/#roles

**Comment by Grafcom** [comment]
Here for example I get lost... yes same handbook... first chapter about plugins...
Rosetta site and forums
https://make.wordpress.org/polyglots/handbook/rosetta/rosetta-expectations/
https://make.wordpress.org/polyglots/handbook/rosetta/theme-plugin-directories/

**Comment by pedro-mendonca** [comment]
I've mentioned this new platform because it's not just like some 3rd party new system that some may or may not use, like Transifex. It's the official WP one.

I know the WP translate site for a while as the main core translation site. Because for now it was just that, it is clear to me that the translation teams should be organized and coherent.
In the last few monthes WP start to test this platform for their core themes and announced future support for translation of themes and plugins stored in the official WordPress.org.

This is why I've mentioned this as something that we should keep an eye on, not that we should abandon our system here. This Polyglots and the Translator Handbook aren't new. But as this is something that is in a huge process of changing, we should be aware of it.
I am a bit concerned that with this huge change of including everything for one single transation team will explode and reduce dramatically the quality of translations, it's something that we must be aware and try to give inputs and suggestions, as a community. It's moving fast now, probably much will change as it keeps going and showing downfalls.

At the moment, as this new site started, I don't know how it imports the translation files, probably from the releases I would guess, but they get immediatly available for new suggestions, approvals and rejections from the Language Translation Teams. Because this is a huge change in the workflow, and many developers have their own dedicated translators like here in Qtranslate-x, I've seen three different things happening. First, some developers ask on GlotPress Forum for Language Teams admins to give validator permissions to the translators that already work with them, wich will increase the confusion and probably create a huge mess on the Core Translations.
The other situation I've seen is some developers don't have their plugins listed on this platform, I don't know how. Then they take care of the translations in the releases.
Another situation in the middle of these is some developers choose not to use the translations from the platform, by removing some files, don't know how this works because I don't manage any plugin and didn't ever saw these backends as developer. In this last case it's some kind of priority established by authors, wich I hope that are the ones that always keep the control.

I'm hoping that as this grows, it will get a bit more controlled, maybe some permissions by project and not just by Locale. Or at least I hope that authors/developers can decide to use or not this translation mechanism, as some already have established their own GlotPress installs, GitHub contributers like here, or even email. Personally I do prefer Poedit rather than this plattform for many reasons, including string debugging, local cache in end of .po, etc.

Let's not panic on this, and just keep our eyes open to understand what is changing and what might affect our already very nice workflow :)

Now, to respond to your question:
> so I must become a member of the Dutch translation team?
If you want to have the strings immediatly approved, yes.
On the other hand, if your language team is aware and approves quickly, you won't have to worry much. And I think that the .mo files could get updated if compared with the repository ones, wich liberates developers from having it all translated at the moment of release.

https://make.wordpress.org/polyglots/teams/?locale=nl_NL
You might present yourself to the Team and ask for permission, of course this is easier when each other already trust on the others work.

Sorry if this is confusing, this is a bit overwhealming and I only mean to help.
Hope it did :)

**Comment by Grafcom** [comment]
@pedro-mendonca 
>Sorry if this is confusing, this is a bit overwhealming and I only mean to help.
Hope it did :)

Yes you did !! :-)
>I am a bit concerned that with this huge change of including everything for one single transation team will explode and reduce dramatically the quality of translations

I agree, that is why I am critical...
>it's something that we must be aware and try to give inputs and suggestions, as a community

Also agree, but they are not clear for plugin authors. We get an email that the plug-in is included  in this new system. More information? No, even not on their website.
>This Polyglots and the Translator Handbook aren't new. But as this is something that is in a huge process of changing, we should be aware of it.

We now know, thanks to you :-)
Again Pedro thanks, also for all your work for qTranslate X!

**Comment by pedro-mendonca** [comment]
:+1:

**Comment by johnclause** [comment]
So, I guess, let us keep everything as it is now. If someone starts using WP system, then we will accommodate for that person. We do not have too many active translators, everyone does the things in a way, which they like, and that is just fine with me. Thank you all for the input!
